src/python_testing/TC_MCORE_FS_1_3.py

```diff
@@ -0,0 +1,181 @@
+
+        discriminator = 3840
+        device_for_dut_ecosystem_vendor_id = await self.read_single_attribute_check_success(cluster=Clusters.BasicInformation, attribute=Clusters.BasicInformation.Attributes.VendorID, dev_ctrl=self.TH_server_controller, node_id=self.device_for_dut_eco_nodeid, endpoint=root_node_endpoint)
+        device_for_dut_ecosystem_product_id = await self.read_single_attribute_check_success(cluster=Clusters.BasicInformation, attribute=Clusters.BasicInformation.Attributes.ProductID, dev_ctrl=self.TH_server_controller, node_id=self.device_for_dut_eco_nodeid, endpoint=root_node_endpoint)
+        # To prevent need for runtime QR code generation test makes some assumptions
+        # about the app that it valiudates here.
+        asserts.assert_equal(passcode, 20202021, "Test implementation assumption incorrect for passcode")
+        asserts.assert_equal(device_for_dut_ecosystem_vendor_id, 0xFFF1, "Test implementation assumption incorrect for vendor ID")
+        asserts.assert_equal(device_for_dut_ecosystem_product_id, 0x8001, "Test implementation assumption incorrect for product ID")
+
+        self.TH_server_controller.OpenCommissioningWindow(nodeid=self.device_for_dut_eco_nodeid, timeout=900, iteration=1000,
+                                                          discriminator=discriminator, option=1)
+
+        setup_qrcode = "MT:-24J0-Q000KA0648G00"
+        self.wait_for_user_input(
+            prompt_msg=f"Using the DUT vendor's provided interface, commission the device using the following parameters:\n"
+            f"- discriminator: {discriminator}\n"
+            f"- passcode: {passcode}\n"
+            f"- setupPinCode: {setup_qrcode}\n"
+            f"If using FabricSync Admin, you may type:\n"
+            f">>> pairing onnetwork 111 {setup_qrcode}")
+
+        root_part_list = await self.read_single_attribute_check_success(cluster=Clusters.Descriptor, attribute=Clusters.Descriptor.Attributes.PartsList, endpoint=root_node_endpoint)
+        set_of_endpoints_after_adding_device = set(root_part_list)
+
+        asserts.assert_true(set_of_endpoints_after_adding_device.issuperset(
+            set_of_endpoints_before_adding_device), "Expected only new endpoints to be added")
+        unique_endpoints_set = set_of_endpoints_after_adding_device - set_of_endpoints_before_adding_device
+        asserts.assert_equal(len(unique_endpoints_set), 1, "Expected only one new endpoint")
+        newly_added_endpoint = list(unique_endpoints_set)[0]
+
+        th_sed_dut_unique_id = await self.read_single_attribute_check_success(cluster=Clusters.BridgedDeviceBasicInformation, attribute=Clusters.BridgedDeviceBasicInformation.Attributes.UniqueID, endpoint=newly_added_endpoint)
+        asserts.assert_true(type_matches(th_sed_dut_unique_id, str), "UniqueID should be a string")
+        asserts.assert_true(th_sed_dut_unique_id, "UniqueID should not be an empty string")
```

Comment:
Isn't this read supposed to fail with `UNSUPPORTED_ATTRIBUTE` instead of returning empty string value when the unique id is absent?

Reply:
The server see this as a string to encode, and can encode an empty string, we as a client might get an empty string here from a misbaving device.

I will double check in a hacked test locally, but I see no harm in having this addtional check here. If we get a `UNSUPPORTED_ATTRIBUTE` that will have an exception raised on ` self.read_single_attribute_check_success` above

Reply:
Confirm a DUT could theoretically send an empty string we should say it is not valid as there is no way that is unique